Add docstring for findAxis function

felixcremer opened the pull request.

meggart commented on src/Cubes/Axes.jl:
```diff
@@ -293,7 +293,10 @@ Given the string of an axis name and a cube, returns this axis of the cube.
-"Fallback method"
+"""
+    findAxis(desc, c)
+Given an Axis description and a cube return the index of the Axis.
```
Maybe we could quickly summarize what an axis description can be (Axis name, abbreviated axis name, an axis itself...)? Would be more clear

felixcremer replied:
I started a list. 
Could you explain how the ByFunction description is used?
Because I haven't found an example and haven't used it myself yet.

meggart replied:
I think we can remove this one. It was necessary to have when we were still registering functions, but it is obsolete now.

meggart replied:
I mean we can remove everything connected to "ByFunction"

felixcremer replied:
I'll open a new PR for that.

felixcremer replied:
See #108

felixcremer commented:
I added the available axis descriptions to the findAxis and getAxis docstrings. 
I also added a docstring for the Axes submodule. 
This uses DocStringExtensions to provide a list of the exports. 

I also want to start documenting the internals of the package and thought about writing Internal docstrings to also clarify my understanding. 
The Internal docs would look like that:
```julia
help?> Axes.get_descriptor
  Internal
  ≡≡≡≡≡≡≡≡≡≡

  get_descriptor(a)

  Get the descriptor of an Axis. This is used to dispatch on the descriptor.
"""
